### PR TITLE
upgrading TestGameState to use card types

### PR DESCRIPTION
**Pull request by kuwizard**
Just a little improvement :)

### CODE REVIEW

File: modules/GT_TestGameState.php [hunk 1]
```diff
@@ -3,12 +3,28 @@
...
 
+use GT\Managers\CardsManager;
+
 class GT_TestGameState
 {
   public function __construct($game, $players)
   {
     $this->game = $game;
     $this->players = $players;
+    $this->cardsCount = [
```
Comment (pickardj79) [inline modules/GT_TestGameState.php:14]: Can this be called `cardsInstanceIndex` instead? or `cardsInstIdx`? These numbers are indexes into the instance array within the card type. I think `cardCount` implies something else, like the count of each type of card?

File: modules/GT_TestGameState.php [hunk 2]
```diff
@@ -62,17 +78,12 @@ function prepareFlight($nextState)
...
-    // $this->setCardOrder(4,1); // openspace card
-    // $this->setCardOrder(17,1); // abship card
-    // $this->setCardOrder(18,1); // abstation card
-    // $this->setCardOrder(0,1); // slavers card
-    $this->setCardOrder(1, 1); // smugglers card
-    $this->setCardOrder(2, 1); // pirates card
-    $this->setCardOrder(15, 1); // combat zone card
+    // Usage: [CARD_PIRATES, CARD_PLANETS, CARD_SMUGGLERS] - they will appear in this order in the game
+    // Unfortunately works a bit problematic with the same type given 2+ times
+    $cards = [CARD_COMBAT_ZONE, CARD_PIRATES, CARD_SMUGGLERS];
```
Comment (pickardj79) [inline modules/GT_TestGameState.php:83]: These are `$card_types`, right?

File: modules/Managers/CardsManager.php [hunk 3]
```diff
@@ -37,14 +35,20 @@ public static function getTypeById($id) {
...
   public static function get($id) {
     $cardType = self::getTypeById($id);
     $className = 'GT\Cards\\'.self::$classes[$cardType];
-    $params = self::getInstance($className, $id);
+    $params = self::getInstanceById($className, $id);
     return new $className($params);
   }
+
+  // Used for TestGameState only at the moment
+  public static function getInstanceIdByType($cardType, $count = 0) {
```
Comment (pickardj79) [inline modules/Managers/CardsManager.php:50]: Similarly, I think of this second argument as an `$index`

**Review by pickardj79 — CHANGES_REQUESTED** [review]

**Comment by pickardj79** [comment]
Great change, btw!

File: modules/GT_TestGameState.php (hunk 1 above)
Comment (kuwizard) [reply modules/GT_TestGameState.php:14]: I don't like shortening in variables names but yes, cardsInstanceIndex makes sense

**Comment by kuwizard** [comment]
@pickardj79, changed everything mentioned in your comments

**Review by pickardj79 — APPROVED** [review]
awesome

**Pull request merged**